tests/Microsoft.Identity.Test.Unit/PublicApiTests/AccountIdTest.cs

```diff
@@ -34,5 +34,75 @@ public void AccountIdPublicApi()
            new AccountId("a.b", "a", "b");
            new AccountId("adfs");
         }
+
+        [TestMethod]
+        [Description("Test the parsing of HomeAccountId with no dot")]
+        public void ParseFromStringNoDotTest()
+        {
+            AccountId accountId = AccountId.ParseFromString("uid");
+
+            Assert.AreEqual("uid", accountId.ObjectId);
+            Assert.AreEqual(null, accountId.TenantId);
+        }
```

Comment:
The tests can be condensed into one test with adding `[DataRow]` attribute like in https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/cfa2b36b9b5732d182bf6e31bb8a9fbc1127ab3a/tests/Microsoft.Identity.Test.Unit/AppConfigTests/PublicClientApplicationBuilderTests.cs#L383